### ISSUE TITLE
Support many to many

### ISSUE THREAD

**Issue by jschoch** [elixir-lang/ecto#183]
does ecto support many to many?  If so maybe a section on the main page on how this works, the tests for associations doesn't seem to test this case.

**Comment by josevalim** [comment]
This depends on #357. We need to have a mechanism for updating associations on Repo.insert/update/delete before we start tackling this.

**Comment by jontonsoup** [comment]
Looking through the docs and noticed that this issue might be outdated? Re: http://hexdocs.pm/ecto/Ecto.Schema.html

**Comment by taybin** [comment]
I looked and didn't see anything mentioning many-to-many. Was there something that jumped out at you?

**Comment by jontonsoup** [comment]
has_many :through is a many to many relationship. 

```
In the example above, we have defined a has_many :through association named :comments_authors. A :through association always expect a list and the first element of the list must be a previously defined association in the current module. For example, :comments_authors first points to :comments in the same module (Post), which then points to :author in the next model Comment.

This :through associations will return all authors for all comments that belongs to that post:
```

**Comment by josevalim** [comment]
@jontonsoup indeed.

We also want to support more direct many to many which automatically manages the intermediate table. And that's what this issue is about. :)

**Comment by jontonsoup** [comment]
@josevalim I'm slightly confused. The essential functionality that @patrickdet 's comment describes on Mar 6, 2014 seems to have been implemented according to the docs, although not fully. I must be missing something?

**Comment by josevalim** [comment]
@jontonsoup the functionality described by @patrickdet is not the one aimed to be supported by this issue. We want to add `many_to_many "tags", Tag` or something similar.

**Comment by jontonsoup** [comment]
Ah, I see — thanks for the clarification. 




Best,

Jon

**Comment by chowza** [comment]
Any solutions for many to many relationships?

http://stackoverflow.com/questions/32900114/many-to-many-relationship-in-ecto

**Comment by josevalim** [comment]
I've opened a new issue to propose this feature as this already contains a long discussion. See #1169.
